lib/cloudflare_protected_email_address.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'cgi'
+
+class CloudflareProtectedEmailAddress
+  def initialize(obfuscated)
+    @obfuscated = obfuscated
+  end
+
+  def human_readable
+    return CGI.unescape(escaped_email_address) if CGI.unescape(escaped_email_address).include?('@')
+    obfuscated
+  end
+
+  private
+
+  attr_accessor :obfuscated
+
+  def obfuscated_characters
+    obfuscated.scan(/.{2}/).drop(1).map(&:hex)
+  end
+
+  def key
+    obfuscated[0..1].hex
+  end
```

Comment:
`key` and `obfuscated_characters` here are both taking responsibility for the same thing, but doing it in slightly different ways. This would be cleaner and more maintainable if those both used an underlying method that took care of the conversion e.g. 

```
  def hex_pairs
    obfuscated.scan(/.{2}/).map(&:hex)
  end

  def obfuscated_characters
    hex_pairs.drop(1)
  end

  def key
    hex_pairs.first
  end
```

(not sure if `hex_pairs` is the right name here)